src/Microsoft.SqlTools.ServiceLayer/Connection/SqlConnectionFactory.cs

```diff
@@ -24,5 +26,36 @@ public DbConnection CreateSqlConnection(string connectionString, string azureAcc
             RetryPolicy commandRetryPolicy = RetryPolicyFactory.CreateDefaultConnectionRetryPolicy();
             return new ReliableSqlConnection(connectionString, connectionRetryPolicy, commandRetryPolicy, azureAccountToken);
         }
+        /// <summary>
+        /// Safely casts DbConnection into SqlConnection
+        /// </summary>
+        public SqlConnection ConvertDBConnectionToSqlConnection(DbConnection connection)
+        {
+            // Get a connection to the database for SMO purposes
```

Comment:
This method itself has nothing to do with SMO so I would just omit the comments here mentioning SMO (or at least make them more generic) - it's up to the callers what they do with the value returned.